rootex/core/renderer/shader.h

```diff
@@ -20,31 +27,23 @@ class Shader
 
 	friend class ShaderLibrary;
 
+
 public:
 	virtual ~Shader();
 
-	enum class Type
-	{
-		Vertex,
-		Pixel
-	};
-
-	enum class VertexConstantBufferType
-	{
-		Model,
-		ModelInverse,
-		View,
-		Projection
-	};
-
 	virtual void bind() const;
 	void unbind() const;
-
-	void set(const VertexConstantBufferType& type, const Matrix& constantBuffer);
-	void set(const PSDiffuseConstantBuffer& constantBuffer) const;
-	void set(const PSSolidConstantBuffer& constantBuffer);
 };
 
+class ColorShader : public Shader
+{
+	ColorShader(const LPCWSTR& vertexPath, const LPCWSTR& pixelPath, const BufferFormat& vertexBufferFormat);
+	ColorShader(ColorShader&) = delete;
+	~ColorShader() = default;
+
+	friend class ShaderLibrary;
+
```

Comment:
Shouldn't there be a set(Color) function here similar to [here](https://github.com/sdslabs/Rootex/pull/136/files#diff-d4a570dc07266a99ae0bd9f70f585311R60)?

Reply:
Color VC se milta tha pehle and ab MVC store kar tha most probably

Reply:
But how does it set a color if the shader doesn't have a setColor function?

Reply:
Color is set in Constant Buffer so it the relevant functions lie with Material